src/Couchbase.Lite.Shared/API/Sync/Replicator.cs

```diff
@@ -492,10 +492,10 @@ private void OnDocEndedWithConflict(List<ReplicatedDocument> replications)
                         replication = replication.ClearError();
                     } catch (Exception e) {
                         C4Error error;
-                        if (e.GetBaseException() is CouchbaseLiteException) {
-                            var errCode = ((CouchbaseLiteException)e).Error;
+                        if (e.GetBaseException() is CouchbaseException) {
+                            var errCode = ((CouchbaseException)e).Error;
                             error = new C4Error((C4ErrorCode)errCode);
```

Comment:
This has taken a step backwards.  This will end up reporting incorrect error codes.  I didn't want to outright suggest code blocks but perhaps you should try this:

Make `ReplicatedDocument.Error` settable interally

```
} catch (CouchbaseException e) {
    replication.Error = e;
} catch (Exception e) {
    replication.Error = new CouchbaseLiteException(C4ErrorCode.UnexpectedError, e.Message, e);
}
```

Then you can put the `ReplicatedDocument` constructors back the way they were.